### PR TITLE
Single product page price updated

### PR DESCRIPTION
**Pull request by kurucaner**
Problem:
On the individual product pages, the displayed price doesn't reflect the true price of the selected variant.

Reproduction:
Navigate to: [ACME Drawstring Bag (Black, Size: 9 x 12 inch)](https://demo.vercel.store/product/acme-drawstring-bag?color=Black&size=9+x+12+inch). The price shown is $12, yet when added to the cart, the actual cost is $9.

Solution:
This PR aims to fix this discrepancy by dynamically updating the displayed price based on the chosen product variant.

### CODE REVIEW

File: components/product/product-description.tsx [hunk 1]
```diff
@@ -1,31 +1,51 @@
+'use client'
```
Comment (BC-krasnoshapka) [inline components/product/product-description.tsx:1]: I think changing whole `product-description` to client component has negative consequences.
Alternatively can we transfer state between `VariantSelector` and `AddToCart` via URL params and `useSearchParams`?

Comment (kurucaner) [reply components/product/product-description.tsx]: Thank you for the heads-up. I've removed 'use client' in the product-description. Instead, I've introduced a useSelectedVariant hook which is now being used in both the Price and AddToCart components. I hope this aligns with the proposed approach.

**Comment by MurusMe** [comment]
Thanks, @kurucaner you're a wizard! It really helped me out. RetroFlynn.com, the changes you made really worked for me. Thanks!

**Review by BC-krasnoshapka — APPROVED** [review]

**Comment by krzywiecki** [comment]
What happened to this PR. Prices for specifi variants still don't seem to work.

**Pull request closed without merging**